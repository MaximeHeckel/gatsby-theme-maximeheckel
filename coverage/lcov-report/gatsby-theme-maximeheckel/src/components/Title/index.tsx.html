<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for gatsby-theme-maximeheckel/src/components/Title/index.tsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">gatsby-theme-maximeheckel/src/components/Title</a> index.tsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import styled from '@emotion/styled';
import { withTheme } from 'emotion-theming';
import { Theme } from '../../theme_light';
&nbsp;
const Title = styled.h1`
  color: ${(props: { theme: Theme }) =&gt; props.theme.fontColor};
  font-size: 36px;
  margin-bottom: 25px;
  font-weight: 500;
`;
&nbsp;
export default withTheme(Title);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Aug 05 2019 23:39:07 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
