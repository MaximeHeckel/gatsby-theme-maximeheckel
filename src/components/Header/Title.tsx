import { motion } from 'framer-motion';
import React from 'react';
import AnchorLink from 'react-anchor-link-smooth-scroll';
import styled from '../../utils/styled';
import { HeaderContext } from './Context';

export const TitleWrapper = styled.div`
  overflow: hidden;
  margin-left: 33px;

  h3 {
    margin-bottom: 0px;
  }

  a {
    color: ${(props) => props.theme.fontColor};
    display: block;
    text-decoration: none;
    overflow: hidden;
    max-width: 600px;
    white-space: nowrap;
    text-overflow: ellipsis;
  }

  @media (max-width: 1000px) {
    a {
      max-width: 450px;
    }
  }

  @media (max-width: 900px) {
    a {
      max-width: 300px;
    }
  }

  @media (max-width: 700px) {
    h3 {
      display: none;
    }
  }
`;

const variants = {
  hide: {
    y: 80,
  },
  show: {
    y: 0,
  },
};

export interface HeaderTitleProps {}

export const Title: React.FC<HeaderTitleProps> = (props) => {
  const { collapsed, sticky } = React.useContext(HeaderContext);

  return (
    <TitleWrapper>
      {props.children ? (
        <div data-testid="header-title">
          {sticky ? (
            <motion.div
              initial="hide"
              variants={variants}
              animate={collapsed ? 'show' : 'hide'}
              transition={{ ease: 'easeInOut', duration: 0.5 }}
            >
              <h3>
                <AnchorLink offset="150" href="#top">
                  {props.children}
                </AnchorLink>
              </h3>
            </motion.div>
          ) : null}
        </div>
      ) : null}
    </TitleWrapper>
  );
};
