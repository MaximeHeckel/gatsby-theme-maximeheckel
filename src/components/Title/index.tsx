import styled from '@emotion/styled';
import { withTheme } from 'emotion-theming';
import { Theme } from '../../theme_light';

const Title = styled.h1`
  color: ${(props: { theme: Theme }) => props.theme.fontColor};
  font-size: 36px;
  margin-bottom: 25px;
  font-weight: 500;
`;

export default withTheme(Title);
