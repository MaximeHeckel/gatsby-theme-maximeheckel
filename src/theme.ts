const theme = {
  light: {
    backgroundColor: 'var(--maximeheckel-colors-body-0, #F6F7F8)',
    borderColor: '#f5f5f9',
    boxShadow: '0 0px 12px -6px rgba(0, 24, 40, 0.25)',
    colors: {
      blue: 'var(--maximeheckel-colors-brand, #5184f9)',
      gray: '#F8F8F9',
      pink: '#FED5D7',
      prism: {
        token: `#fff`,
        // tslint:disable-next-line:object-literal-sort-keys
        languageJavascript: `#e8696b`,
        javascript: `#e8696b`,
        background: `#292c34`,
        comment: `#5e6a76`,
        string: `#a8e2a8`,
        var: `#b3bac5`,
        number: `#e4854d`,
        constant: `#b3bac5`,
        plain: `#fff`,
        doctype: `#e8696b`,
        tag: `#e8696b`,
        keyword: `#d49fd4`,
        boolean: `#ff5874`,
        function: `#5F8DC3`,
        parameter: `#F9965D`,
        className: `#ffcf74`,
        attrName: `#bf87ba`,
        attrValue: `#a8e2a8`,
        interpolation: `#fff`,
        punctuation: `#5FA8AA`,
        ['maybe-class-name']: `#fff`,
        property: `#80cbc4`,
        propertyAccess: `#fff`,
        namespace: `#b2ccd6`,
        highlight: `rgba(255,255,255,0.07)`,
        highlightBorder: '#196FD8',
        dom: `#5F8DC3`,
        operator: `#5FA8AA`,
      },
      white: '#FFFFFF',
    },
    fontColor: 'var(--maximeheckel-colors-typeface-0, #2B2D3E)',
    foregroundColor:
      'var(--maximeheckel-colors-body-1, rgba(217, 230, 247, 0.55))',
    overlayBackground:
      'var(--maximeheckel-colors-body-2, rgba(236, 236, 236, 0.8))',
    transitionTime: 0.5,
  },

  dark: {
    backgroundColor: 'var(--maximeheckel-colors-body-0, #1b1e21)',
    borderColor: 'rgba(255, 255, 255, 0.1)',
    boxShadow: 'rgba(0,0,0,0.2) 0px 20px 40px',
    colors: {
      black: '#000000',
      blue: 'var(--maximeheckel-colors-brand, #5184f9)',
      gray: '#F8F8F9',
      pink: '#FED5D7',
      prism: {
        token: `#fff`,
        // tslint:disable-next-line:object-literal-sort-keys
        languageJavascript: `#e8696b`,
        javascript: `#e8696b`,
        background: `#0F111B`,
        comment: `#5e6a76`,
        string: `#a8e2a8`,
        var: `#b3bac5`,
        number: `#e4854d`,
        constant: `#b3bac5`,
        plain: `#fff`,
        doctype: `#e8696b`,
        tag: `#e8696b`,
        keyword: `#d49fd4`,
        boolean: `#ff5874`,
        function: `#5F8DC3`,
        parameter: `#F9965D`,
        className: `#ffcf74`,
        attrName: `#bf87ba`,
        attrValue: `#a8e2a8`,
        interpolation: `#fff`,
        punctuation: `#5FA8AA`,
        ['maybe-class-name']: `#fff`,
        property: `#80cbc4`,
        propertyAccess: `#fff`,
        namespace: `#b2ccd6`,
        highlight: `rgba(255,255,255,0.07)`,
        highlightBorder: '#3c83da',
        dom: `#5F8DC3`,
        operator: `#5FA8AA`,
      },
      white: '#FFFFFF',
    },
    fontColor: 'var(--maximeheckel-colors-typeface-0, #FFFFFF)',
    foregroundColor:
      'var(--maximeheckel-colors-body-1, rgba(14, 20, 27, 0.55))',
    overlayBackground: 'var(--maximeheckel-colors-body-2, rgba(0,0,0,0.40))',
    transitionTime: 0.5,
  },
};

export default theme;
