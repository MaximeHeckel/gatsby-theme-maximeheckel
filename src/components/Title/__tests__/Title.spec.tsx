import { cleanup, render } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import React from 'react';
import Title from '../';
import theme from '../../../theme_light';

describe('Title', () => {
  beforeEach(cleanup);

  it('renders the Title', () => {
    const component = render(
      <ThemeProvider theme={theme}>
        <Title>Test</Title>
      </ThemeProvider>
    );
    expect(component.baseElement).toMatchSnapshot();
  });
});
