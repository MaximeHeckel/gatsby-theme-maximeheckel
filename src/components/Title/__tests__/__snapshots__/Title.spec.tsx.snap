// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title renders the Title 1`] = `
.emotion-0 {
  color: #2B2D3E;
  font-size: 36px;
  margin-bottom: 25px;
  font-weight: 500;
}

<body>
  <div>
    <h1
      class="emotion-0"
    >
      Test
    </h1>
  </div>
</body>
`;
